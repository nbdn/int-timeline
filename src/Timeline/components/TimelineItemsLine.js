// @flow
import React from 'react';

type Props = {
  children: any,
  start: number,
  end: number,
  unitWidth: number,
  lineIndex: number,
  linesCount: number
};

const TimelineItemsLine = (props: Props) => {
  const { children, start, end, unitWidth, lineIndex, linesCount } = props;
  return (
    <div
      className="TimelineItemsGroup__Line"
      key={`timeline_line_${lineIndex}`}
      style={{
        width: end * unitWidth - start * unitWidth,
        height: `${100 / linesCount}%`
      }}
    >
      {children}
    </div>
  );
};

TimelineItemsLine.defaultProps = {
  children: [],
  start: 0,
  end: 0,
  unitWidth: 0,
  lineIndex: 0,
  linesCount: 0
};

export default TimelineItemsLine;
