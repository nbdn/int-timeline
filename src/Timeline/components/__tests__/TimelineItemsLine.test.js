import React from 'react';
import { mount, shallow } from 'enzyme';
import TimelineItemsLine from '../TimelineItemsLine';

import { timelineDataMock, timelineRuleDataMock } from '../../__tests__/mocks';

describe('TimelineItemsLine component test', () => {
  let wrapper;
  let emptyPropsWrapper;
  let groupMock;
  beforeEach(() => {
    groupMock = timelineDataMock[0];
    wrapper = shallow(
      <TimelineItemsLine
        start={groupMock.start}
        end={groupMock.end}
        unitWidth={timelineRuleDataMock.unitWidth}
        lineIndex={0}
        linesCount={groupMock.items.length}
      />
    );
    emptyPropsWrapper = shallow(<TimelineItemsLine />);
  });

  describe('component rendering right', () => {
    it('render props right match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('render with empty props match snapshot', () => {
      expect(emptyPropsWrapper).toMatchSnapshot();
    });
  });

  describe('component has right props', () => {
    it('has right default props', () => {
      const wrapper = mount(<TimelineItemsLine />);
      expect(wrapper.prop('start')).toEqual(0);
      expect(wrapper.prop('end')).toEqual(0);
      expect(wrapper.prop('unitWidth')).toEqual(0);
      expect(wrapper.prop('lineIndex')).toEqual(0);
      expect(wrapper.prop('linesCount')).toEqual(0);
    });

    it('has right given props', () => {
      const wrapper = mount(
        <TimelineItemsLine
          start={groupMock.start}
          end={groupMock.end}
          unitWidth={timelineRuleDataMock.unitWidth}
          lineIndex={0}
          linesCount={groupMock.items.length}
        />
      );
      expect(wrapper.prop('start')).toEqual(groupMock.start);
      expect(wrapper.prop('end')).toEqual(groupMock.end);
      expect(wrapper.prop('unitWidth')).toEqual(timelineRuleDataMock.unitWidth);
      expect(wrapper.prop('lineIndex')).toEqual(0);
      expect(wrapper.prop('linesCount')).toEqual(groupMock.items.length);
    });
  });

  describe('component structure', () => {
    it('has right wrapper class name', () => {
      expect(wrapper.hasClass('TimelineItemsGroup__Line')).toBeTruthy();
    });
  });
});
