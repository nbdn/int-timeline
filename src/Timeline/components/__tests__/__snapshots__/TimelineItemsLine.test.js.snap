// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineItemsLine component test component rendering right render props right match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimelineItemsLine
    end={50}
    lineIndex={0}
    linesCount={1}
    start={5}
    unitWidth={150}
>
    
</TimelineItemsLine>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "timeline_line_0",
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "TimelineItemsGroup__Line",
      "style": Object {
        "height": "100%",
        "width": 6750,
      },
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "timeline_line_0",
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "TimelineItemsGroup__Line",
        "style": Object {
          "height": "100%",
          "width": 6750,
        },
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TimelineItemsLine component test component rendering right render with empty props match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimelineItemsLine
    end={0}
    lineIndex={0}
    linesCount={0}
    start={0}
    unitWidth={0}
>
    
</TimelineItemsLine>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "timeline_line_0",
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "TimelineItemsGroup__Line",
      "style": Object {
        "height": "Infinity%",
        "width": 0,
      },
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "timeline_line_0",
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "TimelineItemsGroup__Line",
        "style": Object {
          "height": "Infinity%",
          "width": 0,
        },
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
